components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java

```diff
@@ -120,16 +129,68 @@ public boolean terminateSessionsByUserId(String userId) throws SessionManagement
     }
 
     @Override
-    public boolean terminateSessionBySessionId(String userId, String sessionId) {
+    public boolean terminateSessionBySessionId(String userId, String sessionId) throws SessionManagementException {
 
-        sessionManagementService.removeSession(sessionId);
-        List<String> sessionIdList = new ArrayList<>();
-        sessionIdList.add(sessionId);
-        UserSessionStore.getInstance().removeTerminatedSessionRecords(sessionIdList);
+        if (isUserAuthorized(userId, sessionId)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Terminating the session: " + sessionId + " which belongs to the user: " + userId + ".");
+            }
+            sessionManagementService.removeSession(sessionId);
+            List<String> sessionIdList = new ArrayList<>();
+            sessionIdList.add(sessionId);
+            UserSessionStore.getInstance().removeTerminatedSessionRecords(sessionIdList);
+            return true;
+        } else {
+            throw handleSessionManagementClientException(SessionMgtConstants.ErrorMessages
+                    .ERROR_CODE_FORBIDDEN_ACTION, userId);
+        }
+    }
 
+    @Override
+    public List<UserSession> getSessionsByUser(User user, int idpId) throws SessionManagementException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Retrieving all the active sessions of user: " + user.getUserName() + " of user store " +
+                    "domain: " + user.getUserStoreDomain() + ".");
+        }
+        return getActiveSessionList(getSessionIdListByUser(user, idpId));
+    }
+
+    @Override
+    public boolean terminateSessionsByUser(User user, int idpId) throws SessionManagementException {
+
+        List<String> sessionIdList = getSessionIdListByUser(user, idpId);
+        if (log.isDebugEnabled()) {
+            log.debug("Terminating all the active sessions of user: " + user.getUserName() + " of user store " +
+                    "domain: " + user.getUserStoreDomain() + ".");
+        }
+        terminateSessionsOfUser(sessionIdList);
+        if (!sessionIdList.isEmpty()) {
+            UserSessionStore.getInstance().removeTerminatedSessionRecords(sessionIdList);
+        }
         return true;
     }
 
+    @Override
+    public boolean terminateSessionBySessionId(User user, int idpId, String sessionId) throws
+            SessionManagementException {
+
+        if (isUserAuthorized(user, idpId, sessionId)) {
```

Comment:
Null check for user, idpId and sessionId and send proper exceptions

Reply:
introduced errors for invalid user and invalid session id.